### DISCUSSION TITLE
SRT caller and listener

### DISCUSSION THREAD

**Discussion by GabeDev** [Haivision/srt#2803] [Q&A]
Hello I am trying to set up a connection between 2 laptops running cent os 7 , one that acts as a Caller and one as listener. Ive tried to navigate thiss task around  to host a UDP session using a multicast source and port and the caller to connect to the Listener and forward the UDP received session to another UDP session using another network interface.

**Comment by ethouris** [comment]
Not sure if I understand.

You want to read from a multicast UDP link, then forward this over SRT to another machine and from there you want to foward it to another UDP link?

Yes, this is easy and you can use `srt-live-transmit` to try this out. These tools should recognize the multicast mask properly, although you can still use `@` before the address to ensure this. For example:

```
srt-live-transmit udp://@239.255.0.1:5555 srt://anotherhost:5001
```

**Reply by GabeDev** [reply]
I want two commands in seperate machine machine1(caller) and machine2(listener)
machine 1 - Listener

Objective: to host a UDP session using a multicast source and port

Multicast source: 239.X.X.XXX

Multicast port: YYYYY

machine 2 Caller

Objective: to connect to the Listener and forward the UDP received session to another UDP session using another network interface

Output Multicast server: 239.X.X.XXX

Output Multicast port: YYYYY

Output Interface IP: 10.75.X.XXX
how can i confirm this will be set up properly,perhaps send some test data between the two machines

**Comment by GabeDev** [comment]
@ethouris for context im able to make a connection with Listener:

Listener
 
Text
srt-live-transmit 'udp://MULTICAST_SOURCE_IP:PORT' 'srt://:SRT_PORT?passphrase=PASSPHRASE&mode=listener' -t:-1 -v

Caller:

Caller
 
Text
srt-live-transmit 'srt://LISTENER_IP?passphrase=PASSPHRASE&latency=200&mode=caller' 

however no traffic is shown by tcpdump

**Reply by GabeDev** [reply]
I sometimes get these error too

16:41:52.768839/srt-live-transm*E:SRT.ea: remove_usock: @991595445 not found as either socket or group. Removing only from epoll system.
Accepted SRT target connection
16:41:53.769045/SRT:GC*E:SRT.ei: epoll/update: IPE: update struck E1 which is NOT SUBSCRIBED to @991595445
16:41:53.769118/SRT:GC*E:SRT.ei: epoll/update: IPE: update struck E1 which is NOT SUBSCRIBED to @991595445
